crates/goose/src/agent.rs

```diff
@@ -221,6 +316,15 @@ impl Agent {
 
         Ok(Box::pin(async_stream::try_stream! {
             loop {
+                
+                let approx_count = TokenCount::from_messages(&system_prompt, &messages, &tools);
```

Comment:
this would be sth like this:
```
let token_counter = TokenCounter::new();
let token_count = token_counter.count_chat_tokens(
    system_prompt,
    &messages,
    &tools,
    Some("gpt-4o"),
);
```